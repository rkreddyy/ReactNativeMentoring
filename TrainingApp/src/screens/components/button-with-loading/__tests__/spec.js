import React from 'react';
import { shallow } from 'enzyme';

import ButtonWithLoading from '../button-with-loading';

describe(`ButtonWithLoading component`, () => {
    const renderComponent = props => shallow(<ButtonWithLoading {...props} />);

    describe('correctly renders', () => {
        it('show loading indicator', () => {
            const component = renderComponent({
                isLoading: true
            });

            expect(component).toMatchSnapshot();
        });

        it('hide loading indicator', () => {
            const component = renderComponent({
                isLoading: false
            });

            expect(component).toMatchSnapshot();
        });

        it('label text', () => {
            const component = renderComponent({
                defaultLabel: "Test button"
            });

            expect(component).toMatchSnapshot();
        });

        it('all props', () => {
            const component = renderComponent({
                isLoading: true,
                isError: false,
                defaultLabel: 'Test Button',
                errorLabel: null
            });

            expect(component).toMatchSnapshot();
        });
    });
});
