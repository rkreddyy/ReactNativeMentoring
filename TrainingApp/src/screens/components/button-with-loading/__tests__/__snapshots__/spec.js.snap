// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithLoading component correctly renders all props 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  hitSlop={
    Object {
      "bottom": 30,
      "left": 20,
      "right": 50,
      "top": 30,
    }
  }
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#148cbf",
      "height": 40,
      "justifyContent": "center",
      "margin": 30,
      "marginLeft": "auto",
      "marginRight": "auto",
      "padding": 8,
      "width": 100,
    }
  }
  underlayColor="#00a8f3"
>
  <LoadingIndicator />
</TouchableHighlight>
`;

exports[`ButtonWithLoading component correctly renders hide loading indicator 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  hitSlop={
    Object {
      "bottom": 30,
      "left": 20,
      "right": 50,
      "top": 30,
    }
  }
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#148cbf",
      "flex": 1,
      "height": 40,
      "justifyContent": "center",
      "margin": 30,
      "padding": 8,
    }
  }
  underlayColor="#00a8f3"
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 18,
        "textTransform": "uppercase",
      }
    }
  />
</TouchableHighlight>
`;

exports[`ButtonWithLoading component correctly renders label text 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  hitSlop={
    Object {
      "bottom": 30,
      "left": 20,
      "right": 50,
      "top": 30,
    }
  }
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#148cbf",
      "flex": 1,
      "height": 40,
      "justifyContent": "center",
      "margin": 30,
      "padding": 8,
    }
  }
  underlayColor="#00a8f3"
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 18,
        "textTransform": "uppercase",
      }
    }
  >
    Test button
  </Text>
</TouchableHighlight>
`;

exports[`ButtonWithLoading component correctly renders show loading indicator 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  delayPressOut={100}
  hitSlop={
    Object {
      "bottom": 30,
      "left": 20,
      "right": 50,
      "top": 30,
    }
  }
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#148cbf",
      "height": 40,
      "justifyContent": "center",
      "margin": 30,
      "marginLeft": "auto",
      "marginRight": "auto",
      "padding": 8,
      "width": 100,
    }
  }
  underlayColor="#00a8f3"
>
  <LoadingIndicator />
</TouchableHighlight>
`;
